crates/bls_utils/src/verification.rs

```diff
@@ -63,6 +65,14 @@ pub fn get_index_in_commitments(
     )))
 }
 
+pub fn to_g1_affine(pubkey: &dvt_abi::BLSPubkey) -> G1Affine {
+    G1Affine::from_compressed(&pubkey).into_option().unwrap()
```

Comment:
As mentioned before, an `unwrap()` in the code is a code smell.

In this case, you could replace it with an `expect()`, which is the same, but gives context why the unwrap failed:

```rust
    G1Affine::from_compressed(&pubkey)
        .into_option()
        .expect("G1 point is not torsion free.")
```

I inferred this from reading the source of `from_compressed()`, but perhaps the message could be better.